cmd/signimage/signimage.go

```diff
@@ -0,0 +1,309 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"flag"
+	"fmt"
+	"github.com/docker/cli/cli/config"
+	dockertypes "github.com/docker/cli/cli/config/types"
+	"github.com/go-logr/logr"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
+	"io"
+	"k8s.io/klog/v2/klogr"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func checkArg(arg *string, varname string, fallback string) {
+	if *arg == "" {
+		if fallback != "" {
+			*arg = fallback
+		} else {
+			fmt.Printf("%s not found:\n", varname)
+			flag.PrintDefaults()
+			os.Exit(0)
+		}
+	}
+	return
+}
+
+func canonicalisePath(path string) string {
```

Comment:
```suggestion
func canonicalisePath(path string) string {
       filepath.IsAbs(path) {
           return path
       }
       return "/" + filepath.Clean(path)
}
       
```

Reply:
Also, lets add a comment that explains all the various filepaths scenarios that can be defined by the layet

Reply:
I've pushed a variant of this (appending the "/" needs to be inside the clean, also I think its good to canonicalise absolutepaths just in case they have "./" etc in them), I've also changed it so the filestosign list gets canonicalised as well, just in case someone doesn't read the instructions and tries to sign something with a relative path, or adds "../" type things to the path.

Reply:
Kudos for your version, it is shorter :). Regarding filestosign: do we want to try to correct users' mistakes? What if by trying to correct their mistake we will actually sign a file in an incorrect path? I think that in case user has provided incorrect path, we can just print a warning message , and that's it. WDYT?

Reply:
I used to write perl as a job, shorter is not always better!

I agree about chance for signing the wrong file, I think its very unlikely to ever happen but its cleaner just to put out an error message and let the user fix their problem rather than take the risk. I have updated the code.